### PR TITLE
feat(tech): replace deprecated rc-progress

### LINKED ISSUE #105: tech: deprecate `rc-progress` package
Replace `rc-progress` usage with a self-developed progressbar
image
image

### PR DESCRIPTION
**Pull request by chrsi**
A new progress bar component has been introduced that replaces the deprecated rc-progress component. Styling was mostly moved to the component, since it's cut for this concrete use case. Only a single prop is exposed to control the progress.

Fixes #105

### CODE REVIEW

**Comment by chrsi** [comment]
There's still some styling issue when the value is low (eg. 1%)

Screenshot 2023-07-06 at 21 47 27

tbc

**Comment by p4dd9** [comment]
Nice-To-Have: You could add a transition animation for the width of the green/goldish bar so that when the amount is fetched clientside it has some animated loading

**Comment by chrsi** [comment]
> Nice-To-Have: You could add a transition animation for the width of the green/goldish bar so that when the amount is fetched clientside it has some animated loading

Done :)
A possible improvement would be to include the color change in the transition. So its yellow during the transition and green at the end.

**Comment by p4dd9** [comment]
> 

Regarding transition color change at the end of `width` animation --> `transitionend` eventlistener could be used, or  making use of `keyframe` animation where the color changes at the end of the animation at 100% step

File: src/app/(site)/[streamer]/[wishSlug]/components/content.tsx [hunk 1]
```diff
@@ -118,14 +116,6 @@ export const DonatePageContent: NextPage<DonationPageProps> = ({ cms }: Donation
 		setIFrameError(true)
 	}, [])
 
-	useEffect(() => {
```
Comment (p4dd9) [inline src/app/(site)/[streamer]/[wishSlug]/components/content.tsx:121]: nice

**Review by p4dd9 — APPROVED** [review]

**Pull request merged**